### PR TITLE
Verses by competition query

### LINKED ISSUE #33: Create versesByMonth GraphQL Query
versesByMonth query should get info for all verses of a given month

NOTE:
* totalVotes is the number of votes on a track (verse.votes.length)
* upvoted is a boolean that signifies whether the user passed in with userId argument 

Request should be:
query versesByMonth(month: month_integer userId: ID) {
 verseId
 path
 name
 totalVotes
 upvoted
 user { userType }
}

Response should be:
{
 "data": [
 {
    "verseId": <integer>
    "path": <string>,
    "name": <string>,
    "totalVotes": <integer>
    "upvoted": <boolean> (true if user has      voted on this track)
    "user": {
     "id": <integer>
        "artist": <string>,
           "image": <string>,
   }
    }
    }.
 \\ many more verse hashes like this ^
 ]
}

### PR DESCRIPTION
**Pull request by nmking22**
Adds functionality for verses by competition query
* Adds upvoted and voteCount fields to verses
* Connects competitions, verses and users through type fields
* Tests all added functionality

Resolves #33 

## Types of Changes

This PR closes #

- [ ] Bug fix (non-breaking change which fixes an issue): 

- [x] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [x] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

File: app/graphql/types/verse_type.rb [hunk 1]
```diff
@@ -6,6 +6,18 @@ class VerseType < Types::BaseObject
     field :title, String, null: false
     field :competition_id, Integer, null: false
     field :vote, Types::VoteType, null: true
+    field :user, Types::UserType, null: true
+    field :upvoted, Boolean, null: false do
```
Comment (SageOfCode) [inline app/graphql/types/verse_type.rb:10]: Nice, I love the logic on this one.

**Review by SageOfCode — APPROVED** [review]
This all looks great, good work team.

**Review by ETBassist — APPROVED** [review]
Terminated

File: app/graphql/types/verse_type.rb [hunk 2]
```diff
@@ -6,6 +6,18 @@ class VerseType < Types::BaseObject
...
+
+    def upvoted(user_id:)
+      object.votes.where(user_id: user_id) != []
+    end
+
+    def vote_count
+      object.votes.count
+    end
   end
 end
```
Comment (the-color-bliu) [inline app/graphql/types/verse_type.rb:23]: Very cool adding methods into the verse type

**Review by the-color-bliu — APPROVED** [review]
Awesome, merge approved fasho

**Pull request merged**